### PR TITLE
feat(persistence): add task workflow task storage

### PR DESCRIPTION
**Pull request by mushrafmim**
## Summary

Adds persistence support for task workflow task records, including a full orchestration store and a scoped per-task store for restricted subtask access. This supports issue #395 by giving task workflows a durable place to record task ID, macro workflow ID, task template ID, task state, and render data.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Changes Made

- Added the `task_workflow_tasks` migration with `task_id`, `macro_workflow_id`, `task_template_id`, `state`, `data`, and timestamps.
- Added `TaskWorkflowTask` and a GORM-backed `TaskWorkflowStore` for lifecycle-level persistence operations.
- Added `TaskScopedStore`, a restricted wrapper bound to one task ID that only exposes state and data reads/writes.
- Added package documentation describing when to use the full store versus the scoped store.
- Added sqlmock-based tests for the full store and mock-backed tests for the scoped store.

## Testing

- [x] I have tested this change locally
- [x] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

Ran:

```sh
go test ./internal/taskworkflow/...
```

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues

Related to #395

## Screenshots/Demo

Not applicable.

## Additional Notes

The scoped store intentionally wraps the full `Store` interface instead of direct database access. Runtime/orchestration code can hold the full store, while subtasks and render paths can receive only the restricted `TaskScopedStore`.

## Deployment Notes

Run the new database migration before relying on task workflow persistence.

### CODE REVIEW

**Review by sthanikan2000 — COMMENTED** [review]
PR LGTM!
But terminology is bit confusing. 
What about defining the terminologies as follows,
- `MacroWorkflow` -> `Task` 
- `Task` -> `SubTask`

File: backend/internal/taskworkflow/persistence/store.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+// TaskWorkflowTask represents one persisted task within a task workflow.
+type TaskWorkflowTask struct {
+	TaskID          string          `gorm:"type:text;column:task_id;not null;primaryKey" json:"taskId"`
+	MacroWorkflowID string          `gorm:"type:text;column:macro_workflow_id;not null;index" json:"macroWorkflowId"`
+	TaskTemplateID  string          `gorm:"type:text;column:task_template_id;not null;index" json:"taskTemplateId"`
+	State           plugin.State    `gorm:"type:varchar(50);column:state;not null;index" json:"state"`
+	Data            json.RawMessage `gorm:"type:jsonb;column:data;serializer:json;not null" json:"data"`
+	CreatedAt       time.Time       `gorm:"type:timestamptz;column:created_at;not null" json:"createdAt"`
+	UpdatedAt       time.Time       `gorm:"type:timestamptz;column:updated_at;not null" json:"updatedAt"`
```
[the remark this answers is not in the document]
Comment (sthanikan2000) [reply backend/internal/taskworkflow/persistence/store.go]: Address this

**Comment by sthanikan2000** [comment]
> PR LGTM! But terminology is bit confusing. What about defining the terminologies as follows,
> 
> * `MacroWorkflow` -> `Task`
> * `Task` -> `SubTask`

Gone through offline discussion.
Terminologies:
- `SubTask`: Atomic unit; Eg: User Submission, Reviewer Submission
- `Task`: Graph combining `SubTask`; Eg: SimpleForm, WaitForEvent
- `MacroWorkflow`:  Graph combining `Task`; Eg: NPQS workflow, FCAU workflow, Graph combining multiple workflow

File: backend/internal/taskworkflow/persistence/scoped_store.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+type taskScopedStore struct {
+	store  Store
+	taskID string
+}
+
+func NewTaskScopedStore(store Store, taskID string) (TaskScopedStore, error) {
+	if store == nil {
+		return nil, fmt.Errorf("store cannot be nil")
```
Comment (ginaxu1) [inline backend/internal/taskworkflow/persistence/scoped_store.go:38]: can we even get to this point with user passing a `nil` store? wouldn't this be caught with panic at boot time then surfacing here as runtime error?

File: backend/internal/taskworkflow/persistence/store.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+func (s *TaskWorkflowStore) GetByMacroWorkflowID(macroWorkflowID string) ([]TaskWorkflowTask, error) {
+	var tasks []TaskWorkflowTask
+	if err := s.db.Where("macro_workflow_id = ?", macroWorkflowID).Find(&tasks).Error; err != nil {
+		return nil, err
+	}
+	return tasks, nil
+}
+
+func (s *TaskWorkflowStore) Update(task *TaskWorkflowTask) error {
+	return s.db.Save(task).Error
```
Comment (ginaxu1) [inline backend/internal/taskworkflow/persistence/store.go]: nit: better use `.Updates()` to make sure this operates strictly as an update

`.Save()` will update all fields on the struct, even zero-valued ones, and will attempt to insert a new row if the primary key is empty

File: backend/internal/taskworkflow/persistence/scoped_store.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+	return &TaskScopedRecord{
+		TaskID: task.TaskID,
+		State:  task.State,
+		Data:   task.Data,
+	}, nil
+}
+
+func (s *taskScopedStore) GetState() (plugin.State, error) {
+	task, err := s.store.GetByTaskID(s.taskID)
```
Comment (ginaxu1) [inline backend/internal/taskworkflow/persistence/scoped_store.go]: this will do a `SELECT * FROM task_workflow_tasks` just to get the `task.State`. better to add optimized methods to the underlying Store interface, such as `GetStateByTaskID`, using `.Select("state")` to avoid fetching the JSON blob when a plugin only wants to check the state

File: backend/internal/taskworkflow/persistence/store.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+func (s *TaskWorkflowStore) UpdateState(taskID string, state plugin.State) error {
+	return s.db.Model(&TaskWorkflowTask{}).Where("task_id = ?", taskID).Update("state", state).Error
```
Comment (ginaxu1) [inline backend/internal/taskworkflow/persistence/store.go:109]: better check
```go
if res.RowsAffected == 0 {
		return gorm.ErrRecordNotFound
	}
```
in GORM, Update() and .Delete() don't return error if 0 records match; instead, return RowsAffected == 0.

**Review by ginaxu1 — CHANGES_REQUESTED** [review]

**Review by ginaxu1 — APPROVED** [review]
lgtm

**Review by sthanikan2000 — APPROVED** [review]
LGTM!
